### PR TITLE
FILTER_ON_GROUPS_AND_LOCATIONS feature flag now adds a new filter.

### PR DESCRIPTION
**Pull request by pr33thi**
This is work added on to the feature flag `FILTER_ON_GROUPS_AND_LOCATIONS`. 

Before this change: activating the feature flag would change all group + location `OR` logic to `AND` logic. 
This PR changes it so that activating feature flag instead makes a new filter (called `GROUPS_AND_LOCATIONS`) appear, and selecting the filter is what changes `OR`s to `AND`s. The aim of this is to make it clear to the user what is happening (for the case where the feature flag may be on/off and the user isn't getting back the data that they think they are).

This is currently only for the `ONSE` domain, but may be rolled out to more users after talking more with saas.

I also specifically want some help with the wording of the feature flag- is it clear that this is only for form reports/exports, and that it only works if a location and group are both selected (meaning, it doesn't AND together two locations)?

Also, I'm planning to have QA take a look before I merge this.

@esoergel 
code buddy @sravfeyn

### CODE REVIEW

File: corehq/apps/reports/filters/case_list.py [hunk 1]
```diff
@@ -19,10 +19,11 @@ def static_options(self):
         options = super(CaseListFilterUtils, self).static_options
         # replace [Active Mobile Workers] and [Deactivated Mobile Workers] with case-list-specific options
         assert options[0][0] == "t__0"
+        # If static_options include GROUPS_AND_LOCATIONS, the case list should not show it
         return [
             ("all_data", _("[All Data]")),
             ('project_data', _("[Project Data]"))
-        ] + options[2:]
+        ] + options[2:6]
```
Comment (esoergel) [inline corehq/apps/reports/filters/case_list.py]: At this point I'd prefer explicitly defining all static options here, even if it means that they are duplicated.  As-is, it drops the first two options, adds two more, and drops any options beyond the 6th index.  This is fairly obscure, and prone to failure if someone changes the options in the super class.  They could be put in global constants if that's preferable.

File: corehq/apps/reports/models.py [hunk 2]
```diff
@@ -80,16 +80,18 @@ class HQUserType(object):
     COMMTRACK = 4
     DEACTIVATED = 5
     WEB = 6
+    GROUPS_AND_LOCATIONS = 7
```
Comment (esoergel) [inline corehq/apps/reports/models.py]: nit: This isn't really a type of user, it's a filter option, more analogous to "All Data" or "Project Data".  What would you think of defining it like those?

File: corehq/apps/reports/filters/case_list.py (hunk 1 above)
Comment (pr33thi) [reply corehq/apps/reports/filters/case_list.py]: That makes sense, I'm going to do this in a separate PR

Comment (pr33thi) [reply corehq/apps/reports/filters/case_list.py]: Addressed and cleaned up code here: https://github.com/dimagi/commcare-hq/pull/22586

**Comment by czue** [comment]
@esoergel should this still be open?

**Comment by esoergel** [comment]
Nah, I don't think so, unless Product decides to reprioritize.

**Pull request closed without merging**